src/elastic-search.js

```diff
@@ -102,8 +102,15 @@ module.exports = function sendTestResults(testResultsLog, done) {
       });
     }
 
+    var auth = ""
+    if (!!extraParams.elasticsearchPassword === true || !!extraParams.elasticsearchPassword === true) {
+      auth = `${extraParams.elasticsearchUsername}:${extraParams.elasticsearchPassword}@`
+      // Override any username:password in the elasticSearchHost param
+      var hostname = repoConfig.elasticSearchHost.split("@")
```

Comment:
i think here you are assuming that everyone is using elastic cloud with or without credentials but that may not be true for all scenarios. I think something like this might work better

```    var auth = ""
    var hostname = repoConfig.elasticSearchHost
    if (!!extraParams.elasticsearchPassword === true || !!extraParams.elasticsearchPassword === true) {
      auth = `${extraParams.elasticsearchUsername}:${extraParams.elasticsearchPassword}@`
      // Override any username:password in the elasticSearchHost param
      hostname = `${auth}${hostname}`
    }
    var esClient = elasticsearch.Client({
      host: hostname,
      log: currentLogLevel,
      requestTimeout: currentTimeout
    });```

Reply:
You are absolutely right, thx for the snippet.